pkg/ingester/ingester.go

```diff
@@ -700,10 +700,12 @@ func (i *Ingester) QuerySample(req *logproto.SampleQueryRequest, queryServer log
 	if err != nil {
 		return err
 	}
+	var iters []iter.SampleIterator
```

Comment:
Results from these iterators are sent to the querier in batches [here](https://github.com/grafana/loki/blob/c23e541fe7d6aa4e35e720df4e191f034325d24f/pkg/ingester/instance.go#L873). Those batches may end up in different instances of iterators in the querier and result in duplicates even when though we use a `MergeSampleIterator` [here](https://github.com/grafana/loki/blob/c23e541fe7d6aa4e35e720df4e191f034325d24f/pkg/querier/querier.go#L214). 

This change ensures that an instance always uses a `MergeSampleIterator` to avoid sending duplicates in a single stream to the querier.